src/dpg_slam/dpg_slam.cc

```diff
@@ -613,33 +602,106 @@ namespace dpg_slam {
+	ROS_INFO_STREAM("Current Pose Chain Created, size: " << currPoseChain.size());
 
-        std::vector<occupancyGrid> grids{currGrid, subMapGrid};
-        return grids;
+	// create occupancy grids for nodes in current pose chain
+	std::vector<occupancyGrid> currPoseChainGrids;
+	for (uint32_t i=0; i< currPoseChain.size(); i++) {
+	    DpgNode currNode = currPoseChain[i];
+	    currPoseChainGrids.emplace_back(currNode, dpg_parameters_, pose_graph_parameters_);
+	}
+	
+	ROS_INFO_STREAM("Occupancy Grids Created for Current Pose Chain Nodes, size: " << currPoseChainGrids.size());
+
+        // Grid for FOV nodes
+        occupancyGrid localSubMapGrid = getSubMapCoveringCurrPoseChain(currPoseChain, currPoseChainGrids);
+	
+	return std::make_pair(currPoseChainGrids, localSubMapGrid);
     }
     
-    std::vector<DpgNode> DpgSLAM::getNodesCoveringCurrGrid(const occupancyGrid& currGrid) {
-        std::vector<DpgNode> FovNodes;
-        for (uint32_t i=0; i<(dpg_nodes_.size()-current_pass_nodes_.size()); i++) {
-            // TODO: FIll in
-        }
-        return FovNodes;
+    occupancyGrid DpgSLAM::getSubMapCoveringCurrPoseChain(const std::vector<DpgNode> &poseChain, 
+		    						const std::vector<occupancyGrid> &poseChainGrids) {
+	
+	occupancyGrid subMapOccupancyGrid(dpg_parameters_, pose_graph_parameters_);
+	std::unordered_set<cellKey, boost::hash<cellKey>> currentUncoveredCells;
+	bool isLocalSubmapInitialized = false;
+
+	// populate the list of uncovered cells
+	for (uint32_t i=0; i<poseChainGrids.size(); i++) {
+		std::unordered_map<cellKey, CellStatus, boost::hash<cellKey>> currNodeGridInfo = poseChainGrids[i].getGridInfo();
+		for (const auto &kv : currNodeGridInfo) { 
+			currentUncoveredCells.insert(kv.first);
+		}
+	}
+	uint64_t totalUncoveredCells = currentUncoveredCells.size();
+	uint64_t currentUncoveredCellsSize  = currentUncoveredCells.size();
+	
+	// compute submap that covers the currentUncoveredCells
+	// Get max laser range
+	float proximityThreshold = dpg_parameters_.distance_threshold_for_local_submap_nodes_;
+	float coverageThreshold = dpg_parameters_.current_pose_graph_coverage_threshold_;
+
+	for (uint32_t j=0; j<(dpg_nodes_.size()-current_pass_nodes_.size()); j++) { 
+		
+		DpgNode pastNode = dpg_nodes_[j];
+		if (!pastNode.isNodeActive())
+			continue;
+
+                std::pair<Vector2f, float> pastNodeInfo = pastNode.getEstimatedPosition();
+                Vector2f pastNodePosition = pastNodeInfo.first;
+		// checking if the past node is close to any of the current nodes.
+		bool isInProximity = false;
+		for (uint32_t i=0; i<poseChain.size(); i++) {
+			DpgNode currNode = poseChain[i];
+                	std::pair<Vector2f, float> currentNodeInfo = currNode.getEstimatedPosition();
+               		Vector2f currentNodePosition = currentNodeInfo.first;
+			float distanceToPastNode = (currentNodePosition - pastNodePosition).norm();
+			if (distanceToPastNode <= proximityThreshold)
```

Comment:
You still need to set the isInProximity variable, but it can be in the same if right before the break.